eZ/Publish/Core/Persistence/Legacy/User/Role/Gateway/DoctrineDatabase.php

```diff
@@ -47,15 +47,21 @@ public function __construct(DatabaseHandler $handler)
      */
     public function createRole(Role $role)
     {
-        // Role ID is set when creating a draft from an existing role
+        // Role original ID is set when creating a draft from an existing role
+        if ($role->status === Role::STATUS_DRAFT && $role->id) {
+            $roleOriginalId = $role->id;
+        } elseif ($role->status === Role::STATUS_DRAFT) {
+            $roleOriginalId = -1;
+        } else {
+            $roleOriginalId = Role::STATUS_DEFINED;
```

Comment:
For better readability this should either use `0`, or a constant that makes it clear, like `DEFINED_ROLE_EMPTY_ORIGINAL_ID`. Same point goes for use of `-1` above.